appengine/src/test/java/com/google/googleinterns/gscribe/services/ExamServiceTest.java

```diff
@@ -0,0 +1,490 @@
+            valueRangeDataWhole.add(Arrays.asList(new String[]{"Duration of exam:"}));
+            valueRangeDataWhole.add(Arrays.asList(new String[]{"Question Type", "Question statement", "Option A", "Option B", "Option C", "Option D", "Points"}));
+            valueRangeDataWhole.add(Arrays.asList(new String[]{"SUBJECTIVE", "SubjectiveStatement1", "", "", "", "", "2"}));
+            valueRangeDataWhole.add(Arrays.asList(new String[]{"MCQ", "MCQStatement1", "OptionA", "OptionB", "OptionC", "OptionD", "3"}));
+            ValueRange valueRangeWhole = new ValueRange();
+            valueRangeWhole.setValues(valueRangeDataWhole);
+
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A:A")).thenReturn(valueRangeForNumberOfQuestions);
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A1:G4")).thenReturn(valueRangeWhole);
+
+            examService.getExam(examRequest, user);
+        });
+        assertEquals("Exam format Exception: duration not present in B1", exception.getMessage());
+
+    }
+
+    @Test
+    public void EmptyExamTest() {
+
+        ExamFormatException exception = assertThrows(ExamFormatException.class, () -> {
+            List<List<Object>> valueRangeDataColumnA = new ArrayList<>();
+            valueRangeDataColumnA.add(Arrays.asList(new String[]{"Duration of exam:"}));
+            valueRangeDataColumnA.add(Arrays.asList(new String[]{"Question Type"}));
+            ValueRange valueRangeColumnA = new ValueRange();
+            valueRangeColumnA.setValues(valueRangeDataColumnA);
+
+            List<List<Object>> valueRangeDataWhole = new ArrayList<>();
+            valueRangeDataWhole.add(Arrays.asList(new String[]{"Duration of exam:"}));
+            valueRangeDataWhole.add(Arrays.asList(new String[]{"Question Type", "Question statement", "Option A", "Option B", "Option C", "Option D", "Points"}));
+            ValueRange valueRangeWhole = new ValueRange();
+            valueRangeWhole.setValues(valueRangeDataWhole);
+
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A:A")).thenReturn(valueRangeColumnA);
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A1:G2")).thenReturn(valueRangeWhole);
+
+            examService.getExam(examRequest, user);
+        });
+        assertEquals("Exam format Exception: Improper exam template used", exception.getMessage());
+
+    }
+
+    @Test
+    public void wrongDurationFormatTest() {
+
+        ExamFormatException exception = assertThrows(ExamFormatException.class, () -> {
+            List<List<Object>> valueRangeDataWhole = new ArrayList<>();
+            valueRangeDataWhole.add(Arrays.asList(new String[]{"Duration of exam:", "asd"}));
+            valueRangeDataWhole.add(Arrays.asList(new String[]{"Question Type", "Question statement", "Option A", "Option B", "Option C", "Option D", "Points"}));
```

Comment:
Lot of code duplication, can you initialise the common components outside.

Reply:
Changed the code to reduce code duplication.